src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3163,9 +3163,9 @@ private static bool AttributeTypeNamesMatch(Type attributeType, Type reflectedAt
 
                 // Collect task, opcode, keyword and channel information
 #if FEATURE_MANAGED_ETW_CHANNELS && FEATURE_ADVANCED_MANAGED_ETW_CHANNELS
-                foreach (var providerEnumKind in new string[] { "Keywords", "Tasks", "Opcodes", "Channels" })
+                foreach (var providerEnumKind in (ReadOnlySpan<string>)["Keywords", "Tasks", "Opcodes", "Channels"])
```

Comment:
> `(ReadOnlySpan<string>)`

Is the cast in places like this something that we'll be able to omit in future C# versions?

Reply:
Likely. Collection expressions _currently_ don't have a natural type, so they currently can't be used in places where no target type is specified.